python_modules/libraries/dagster-aws/dagster_aws/emr/emr_eks_pyspark_step_launcher.py

```diff
@@ -0,0 +1,574 @@
+        application_name = self._application_name(step_context)
+
+        return DagsterEvent(
+            event_type_value=DagsterEventType.STEP_FAILURE.value,
+            pipeline_name=step_context.dagster_run.pipeline_name,
+            step_key=step_context.step.key,
+            message=f"Spark job for '{application_name}' has failed",
+            event_specific_data=StepFailureData(
+                error=None,
+                user_failure_data=UserFailureData(
+                    label="Spark job error",
+                    metadata_entries=[
+                        MetadataEntry(label=k, value=MetadataValue.text(v))
+                        for k, v in dataclasses.asdict(exc).items()
+                    ],
+                ),
+                error_source=ErrorSource.USER_CODE_ERROR,
+            ),
+        )
+
+    def _step_run_ref(
+        self,
+        step_context: StepExecutionContext,
+    ) -> StepRunRef:
+        """Build a StepRunRef that can be used to run the step
+        in a remote process.
+        """
+        step_run_ref = step_context_to_step_run_ref(
+            step_context,
+        )
+
+        if not self.deploy_local_job_package:
+            # Container has the code, but we need to point Dagster to it
+            step_run_ref = EmrEksPySparkResource._set_step_run_ref_working_directory(step_run_ref)
+
+        step_run_ref = EmrEksPySparkResource._remove_parent_run_id(step_run_ref)
+
+        return step_run_ref
+
+    @staticmethod
+    def _set_step_run_ref_working_directory(
+        step_run_ref: StepRunRef,
+        working_directory: str = "/opt/raptor/pipelines/",  # TODO: do not hardcode
+    ) -> StepRunRef:
+        """Hack for submitting code at a different location in the remote
+        containers as locally.
+        """
+        if isinstance(step_run_ref.recon_pipeline.repository.pointer, FileCodePointer):
+            step_run_ref = step_run_ref._replace(
+                recon_pipeline=step_run_ref.recon_pipeline._replace(
+                    repository=step_run_ref.recon_pipeline.repository._replace(
+                        pointer=ModuleCodePointer(
+                            step_run_ref.recon_pipeline.repository.pointer.module,
+                            step_run_ref.recon_pipeline.repository.pointer.fn_name,
+                            working_directory,
+                        )
+                    )
+                )
+            )
+
+        return step_run_ref
+
+    @staticmethod
+    def _remove_parent_run_id(step_run_ref: StepRunRef) -> StepRunRef:
+        """Hack to remove parent run ID for the remote step run ref.
+
+        As an ephemeral Dagster instance is used, the parent run IDs cause
+        issues as they are not known to the ephemeral instance.
+        """
+        return step_run_ref._replace(
+            dagster_run=step_run_ref.dagster_run._replace(root_run_id=None)._replace(
+                parent_run_id=None
+            )
+        )
```

Comment:
question: is this workaround still needed? there was an issues for "Re-runs"

Reply:
it looks like we can remove it